crates/chain/src/keychain/txout_index.rs

```diff
@@ -588,3 +588,63 @@ where
         .take_while(Result::is_ok)
         .map(Result::unwrap)
 }
+
+pub struct SpkIterator<'a> {
+    next_index: u32,
+    end: u32,
+    descriptor: Cow<'a, Descriptor<DescriptorPublicKey>>,
+    secp: Secp256k1<bitcoin::secp256k1::VerifyOnly>,
+}
```

Comment:
Let's start using this in the actual codebase! Create one (or more) constructors for this and replace the calls to `range_descriptor_spks()`. The other way is to make `range_descriptor_spks()` a constructor of `SpkIterator`.

Reply:
Okay, I see you have a nice constructor below `SpkIterator::new`. We need another one that allows us to set a range.